testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -434,23 +434,20 @@
   )
 end
 
-When(/^I enter the hostname of "([^"]*)" as "([^"]*)"$/) do |host, hostname|
+When(/^I enter the hostname of "([^"]*)" as "([^"]*)"$/) do |host, field|
   system_name = get_system_name(host)
   log "The hostname of #{host} is #{system_name}"
-  step %(I enter "#{system_name}" as "#{hostname}")
-end
-
-When(/^I select the hostname of "([^"]*)" from "([^"]*)"$/) do |host, hostname|
-  case host
-  when 'proxy'
-    # don't select anything if not in the list
-    next if $proxy.nil?
-    step %(I select "#{$proxy.full_hostname}" from "#{hostname}")
-  when 'sle_minion'
-    step %(I select "#{$minion.full_hostname}" from "#{hostname}")
-  when 'build_host'
-    step %(I select "#{$build_host.full_hostname}" from "#{hostname}")
+  step %(I enter "#{system_name}" as "#{field}")
+end
+
+When(/^I select the hostname of "([^"]*)" from "([^"]*)"$/) do |host, field|
+  begin
+    node = get_target(host)
+  rescue
+    # don't try to select a non-existing system
+    return
```

Comment:
Shouldn`t we raise or log something here?

Reply:
```bash
Offenses:

features/step_definitions/navigation_steps.rb:446:3: C: Style/RescueStandardError: Avoid rescuing without specifying an error class.
  rescue
  ^^^^^^
```
RuboCop complains, too :wink:

Reply:
yes I can log something like "Ignoring" will do

as elsewhere, we will ignore this ruby cop :-P

Reply:
Ok but what about just logging this? Then we have some feedback and RuboCop is satisfied. Otherwise we have to adjust the RuboCop todo file to include this.

Reply:
```
 --- a/testsuite/.rubocop_todo.yml
+++ b/testsuite/.rubocop_todo.yml
@@ -560,6 +560,7 @@ Style/RegexpLiteral:
 Style/RescueStandardError:
   Exclude:
     - 'features/step_definitions/api_common.rb'
+    - 'features/step_definitions/navigation_steps.rb'
     - 'features/support/cobbler_test.rb'
 
 Style/SafeNavigation:
 ```
```
--- a/testsuite/features/step_definitions/navigation_steps.rb
+++ b/testsuite/features/step_definitions/navigation_steps.rb
@@ -444,7 +444,7 @@ When(/^I select the hostname of "([^"]*)" from "([^"]*)"$/) do |host, field|
   begin
     node = get_target(host)
   rescue
-    # don't try to select a non-existing system
+    log "#{host} is not deployed, not trying to select its hostname from the list"
     return
   end
   step %(I select "#{node.full_hostname}" from "#{field}")
```

Reply:
Ideally, we should have two steps, or an option `ignoring if missing`, for the proxy case.

Reply:
Done, I added a `if present` option.